include/pybind11/pytypes.h

```diff
@@ -1167,6 +1178,20 @@ class bytes : public object {
             pybind11_fail("Unable to extract bytes contents!");
         return std::string(buffer, (size_t) length);
     }
+
+#ifdef PYBIND11_HAS_STRING_VIEW
+    // NOLINTNEXTLINE(google-explicit-constructor)
+    bytes(std::string_view s) : bytes(s.data(), s.size()) { }
+
+    operator std::string_view() const {
+        char *buffer = nullptr;
+        ssize_t length = 0;
+        if (PYBIND11_BYTES_AS_STRING_AND_SIZE(m_ptr, &buffer, &length))
+            pybind11_fail("Unable to extract bytes contents!");
```

Comment:
`Unable to obtain a string_view for this object!`

?

Reply:
No, I don't think so: if this fails it's because we couldn't get the pointer+length out of the underlying Python `bytes` object.  (I don't know under what situation this could actually fail, though).

Reply:
Got it, thanks!